README.md

```diff
@@ -1,19 +1,152 @@
-# Custom Quiz Moodle Plugin
-__User stories__:
-|As a...|I want to...|so that...|
-|------|--------|-----|
-|evaluator|generate quizzes according to clear specifications|I can make sure that every student will have a similar quiz with different questions|
-|evaluator|choose a source for the questions: already loaded questions, new questions from a file or external database|I can choose what set of questions to use for every quiz|
-|evaluator|to have the option to choose from default templates|I can generate the quiz fast, without specifying all the data|
-|evaluator|choose if I want to use new questions or nor|avoid recently used questions|
-|evaluator|keep the default quiz setting available|set the timer, review options, etc.|
-|evaluator|add tags to my questions|specify the difficulty, topic and other relevant data|
-|evaluator|update the last used date of the questions|avoid reusing them|
-|Moodle administrator|avoid question duplicates|create a backup effectively|
-|Moodle administrator|not to crash the server|all the users can use Moodle|
-|student|have the quiz ready when I click "Attempt"|do not waste the allocated time|
-
-
-__References__
- - [QType Template](https://github.com/marcusgreen/moodle-qtype_TEMPLATE/)
- - [Moodle Docs](https://docs.moodle.org/dev/Question_types)
+# Contents
+
+0. [Introduction](/README.md#introduction)
+1. [Installation](/README.md#installation)
+2. [Adding Questions](/README.md#adding-questions)
+3. [Managing Questions](/README.md#managing-questions)
+4. [Updating local database](/README.md#updating-local-database)
+5. [Downloading questions](/README.md#downloading-questions)
+6. [References](/README.md#references)
+
+# Introduction
+
+Abbreviations: MXML (Moodle XML)
+
+This plugin is used for creating quizzes that are unique for every student and contains questions with specific difficulties and topics chosen by evaluators.
+The selection of the questions is automated based on the options that evaluators choose, so you can have a good distribution of difficulty levels and topics.
+
+We want to choose the questions randomly based on the selected difficulty and topic, and to make sure that all the questions are used evenly.
+In order to do this consistently when reusing the same questions in multiple Moodle instances, we need to keep timestamps of every question last use.
+
+As defined in [Managing Questions](README.md#managing-questions), the `last_used` question tag keeps the time of the question was last use, measured in seconds since the Epoch (00:00:00 UTC, January 1, 1970).
```

Comment:
```suggestion
As defined in [Managing Questions](README.md#managing-questions), the `last_used` question tag keeps the time of the question was last used, measured in seconds since the Epoch (00:00:00 UTC, January 1, 1970).
```